pkg/operator/mon/mon.go

```diff
@@ -337,9 +339,9 @@ func getNodeInfoFromNode(n v1.Node) (*NodeInfo, error) {
 func (c *Cluster) startPods(mons []*monConfig) error {
 	for _, m := range mons {
 		node, _ := c.mapping.Node[m.Name]
-
+		nodeId := getNodeId(node)
```

Comment:
There are more places in this file where `node.Name` is being referenced. Did you ensure that those also work when the Hostname differs from the Name? Im starting to think this is a much bigger undertaking that we previously thought.

Reply:
Actually I think it's even simpler. The Hostname of a kubernetes cluster is populated by kubernetes and therefore it is the only one we should use for node selection. As explained in https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetesiohostname , we can assume its existence as a fact.